modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalArticleLocalServiceImpl.java

```diff
@@ -7580,6 +7581,20 @@ private boolean _equals(
 		return false;
 	}
 
+	private int _findLastNoncharPosition(String str, char c) {
+		if (!str.endsWith(String.valueOf(c))) {
+			return str.length() - 1;
+		}
```

Comment:
Can you use `StringUtil.endsWith(String, char)` to avoid creating an additional string?

Reply:
Also, can you inline this method inside the other one? The code is simple enough that we don't need two private methods.
```
if (Validator.isNull(string) || !string.endsWith(StringPool.SLASH)) {
    return string;
}

int end = string.length();
while ((end > 0) && (string.charAt(end - 1) == CharPool.SLASH)) {
    end--;
}

return string.substring(0, end);
```